src/ruby.ts

```diff
@@ -136,8 +146,9 @@ export class Ruby {
   }
 
   private async activate(ruby: string) {
-    const result = await asyncExec(
-      `${this.shell} -ic '${ruby} --disable-gems -rjson -e "printf(%{RUBY_ENV_ACTIVATE%sRUBY_ENV_ACTIVATE}, JSON.dump(ENV.to_h))"'`,
+    const result = await asyncExec(this.windows ?
+      `${this.shell} ${this.shell_opt_activate}  ${ruby} --disable-gems -rjson -e 'printf(%{RUBY_ENV_ACTIVATE%sRUBY_ENV_ACTIVATE}, JSON.dump(ENV.to_h))' ` :
+      `${this.shell} ${this.shell_opt_activate} '${ruby} --disable-gems -rjson -e "printf(%{RUBY_ENV_ACTIVATE%sRUBY_ENV_ACTIVATE}, JSON.dump(ENV.to_h))"'`,
```

Comment:
What's the difference between the two here? Just the single quotes?

Why can't we use quotes on powershell?

Reply:
Right, i agree, though then ChildProcess.exec will use cmd.exe, which has problems with " (double quotes) (can't really escape them). Thats why i use powershell here.

Reply:
So the problem is not powershell, the problem is cmd.exe. I use poweshell to avoid cmd.exe problems.